internal/authinterceptors/interceptor.go

```diff
@@ -0,0 +1,77 @@
+package authinterceptors
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+type userId string
```

Comment:
Most common is to capitalize initialisms: https://go.dev/wiki/CodeReviewComments#initialisms
```suggestion
type userID string
```